text_extensions_for_pandas/array/tensor_test.py

```diff
@@ -0,0 +1,96 @@
+#
+#  Copyright (c) 2020 IBM Corp.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+import numpy as np
+import numpy.testing as npt
+import pandas as pd
+import unittest
+
+from text_extensions_for_pandas.array.tensor import TensorArray
+
+
+class TensorTest(unittest.TestCase):
+
+    def test_create(self):
+        x = np.ones([5, 2, 3])
+        s = TensorArray(x)
+        self.assertEqual(len(s), 5)
+
+        x = [np.ones([2, 3])] * 5
+        s = TensorArray(x)
+        self.assertEqual(len(s), 5)
+
+        x = [np.ones([2, 3]), np.ones([3, 2])]
+        with self.assertRaises(ValueError):
+            TensorArray(x)
+
+        with self.assertRaises(TypeError):
+            TensorArray("foobar")
+
+    def test_operations(self):
+        x = np.ones([5, 3])
+
+        s1 = TensorArray(x)
+        s2 = TensorArray(x)
+        self.assertTrue(np.all(s1 == s2))
+
+        s2 = TensorArray(x * 2)
+        self.assertTrue(np.all(s1 < s2))
+        self.assertFalse(np.any(s1 > s2))
+
+    def test_repr(self):
+        x = np.array([[1, 2], [3, 4], [5, 6]])
+        expected = """\
```

Comment:
You can use `textwrap.dedent` to make these string literals line up with the code. See https://github.com/frreiss/text-extensions-for-pandas/blob/667ad70ed28440b76497f72df9f654c08d28e9dd/text_extensions_for_pandas/spanner/test_join.py#L69

Reply:
worked great, thanks!